pyxem/generators/subpixelrefinement_generator.py

```diff
@@ -78,8 +82,17 @@ class SubpixelrefinementGenerator():
 
     """
 
-    def __init__(self, dp, vectors):
-        self.dp = dp
+    def __init__(self, dp, vectors, padding=None):
+        if padding:
+            dp_padded = np.zeros((dp.data.shape[0], dp.data.shape[1],
+                                  dp.data.shape[2] + padding, dp.data.shape[2] + padding))
+            dp_padded[:, :, int(padding / 2):dp.data.shape[2] + int(padding / 2), int(padding / 2):dp.data.shape[2] + int(padding / 2)] = dp.data
+            dp_padded = pxm.ElectronDiffraction2D(dp_padded)
+            transfer_signal_axes(dp_padded, dp)
+            transfer_navigation_axes(dp_padded, dp)
+            self.dp = dp_padded
```

Comment:
This is brittle to 4D data and should probably be generalized.

Reply:
It's also slightly dodgy behaviour if you're trying to get really precise peak positions, really such edge peaks should just be thrown away but I understand why that hasn't happened in the segmentation case.

Reply:
Update: This is being to workflow (so will be deleted here)